sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/signature/AbstractKeyVaultKeyLessSignature.java

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.security.keyvault.jca.implementation.signature;
+
+import com.azure.security.keyvault.jca.KeyVaultPrivateKey;
+
+import java.nio.ByteBuffer;
+import java.security.InvalidParameterException;
+import java.security.MessageDigest;
+import java.security.PublicKey;
+import java.security.PrivateKey;
+import java.security.SignatureSpi;
+import java.security.SecureRandom;
+
+/**
+ * KeyVault Signature to key less sign
+ */
+public abstract class AbstractKeyVaultKeyLessSignature extends SignatureSpi {
+
+    // message digest implementation we use for hashing the data
+    protected MessageDigest messageDigest;
+
+    protected String keyId;
```

Comment:
How about moving `keyVaultClient` and `signatureParameters` into this class and init them in a constructor?